lib/ansible/template/__init__.py

```diff
@@ -875,10 +871,7 @@ def template(self, variable, convert_bare=False, preserve_trailing_newlines=True
                 return variable
 
         except AnsibleFilterError:
-            if self._fail_on_filter_errors:
-                raise
-            else:
-                return variable
+            raise
```

Comment:
Looks like we could just remove this whole `try/except`

Reply:
Done. And since that basically means that I would be in `git blame` for the whole `template` method anyway I took this as an opportunity to reduce indentation there in subsequent commit (which I can always remove from the PR if nobody wants to review it ;-).